src/fhirtypes/ElementDefinition.ts

```diff
@@ -1537,7 +1537,7 @@ export class ElementDefinition {
       );
     } else if (type === 'Coding') {
       this.assignFHIRValue(code.toString(), code.toFHIRCoding(), exactly, 'Coding');
-    } else if (type === 'Quantity') {
+    } else if (type === 'Quantity' || type === 'Age') {
```

Comment:
This does work to address the task as it was described, but there may be a way to do this more generally. `Age` is not the only specialization of `Quantity`, there are several others. If you check out the graph of General-Purpose Data types on [this](https://www.hl7.org/fhir/datatypes.html#2.24.0) page, the types that are shown to inherit from `Quantity` with blue arrows will all act similarly to `Age`.

For any value that is a `Quantity` or specialization of `Quantity`, we can assign values using two different syntaxes:
```
* some.quantity.element = 64.0 'a'
```
to set the `value`, `code` and implicitly the `system` (which automatically gets set to http://unitsofmeasure.org, since this is a very common `system` to use), or
```
* some.quantity.element = UCUM#a "years"
```
which sets the `system`, `code`, and `units`. It's a little weird, but each syntax is useful for its own purposes.

Anyway, if you test out the first syntax on an element of type `Age` or `Duration` (another specialization of `Quantity`), you will notice it already works. This is because of code on lines 1180-1191 in ElementDefinition.ts. You will notice that this code does not hard code in the name of the specializations, but rather dynamically checks if a type inherits from `Quantity`, and if it does, we let the `64.0 'a'` syntax be used on that element.

I think a similar dynamic check would be useful on this PR. Instead of checking if the type is `Age`, we can look up the StructureDefinition for the given type, and if that StructureDefinition inherits from `Quantity`, then we can assign the FHIR value to the `Quantity` or specialization of `Quantity` element. The code on line 1180-1191 should be a useful example of how to do this, and also let me know if you want more of an explanation.

Reply:
I was wondering if this was a problem specific to the `Age` type or all of the other `Quantity` specializations, it'd definitely be useful to make this fix more general. I'll implement this next, thanks for the heads up and for the example!